src/ast/values.ts

```diff
@@ -12,3 +32,229 @@ export const UNKNOWN_EXPRESSION: ExpressionEntity = {
+const callsArgReturnsNumber: RawMemberDescription = { value: { returns: UNKNOWN_LITERALS.number, callsArgs: [0] } };
+const returnsString: RawMemberDescription = { value: { returns: UNKNOWN_LITERALS.string, callsArgs: null } };
+
+export const UNKNOWN_OBJECT_EXPRESSION: ExpressionEntity = {
+	reassignPath: () => {},
+	forEachReturnExpressionWhenCalledAtPath: () => {},
+	getValue: () => UNKNOWN_VALUE,
+	hasEffectsWhenAccessedAtPath: path => path.length > 1,
+	hasEffectsWhenAssignedAtPath: path => path.length > 1,
+	hasEffectsWhenCalledAtPath: path => {
+		if (path.length === 1) {
+			const subPath = path[0];
+			return isUnknownKey(subPath) || !objectMembers[subPath];
+		}
+		return true;
+	},
+	someReturnExpressionWhenCalledAtPath: (
+		path: ObjectPath,
+		_callOptions: CallOptions,
+		predicateFunction: SomeReturnExpressionCallback,
+		options: ExecutionPathOptions
+	) => {
+		if (path.length === 1) {
+			const subPath = path[0];
+			return isUnknownKey(subPath)
+				|| !objectMembers[subPath]
+				|| predicateFunction(options)(objectMembers[subPath].returns);
+		}
+		return true;
+	},
+	toString: () => '[[UNKNOWN OBJECT]]'
+};
+
+export const arrayMembers: MemberDescriptions =
+	assembleMemberDescriptions({
+		concat: returnsArray,
+		every: callsArgReturnsBoolean,
+		filter: callsArgReturnsArray,
+		find: callsArgReturnsUnknown,
+		findIndex: callsArgReturnsNumber,
+		forEach: callsArgReturnsUnknown,
+		includes: returnsBoolean,
+		indexOf: returnsNumber,
+		join: returnsString,
+		lastIndexOf: returnsNumber,
+		map: callsArgReturnsArray,
+		reduce: callsArgReturnsUnknown,
+		reduceRight: callsArgReturnsUnknown,
+		slice: returnsArray,
+		some: callsArgReturnsBoolean,
+		toLocaleString: returnsString,
+		toString: returnsString
+	});
+
+const literalMembers: { [key: string]: MemberDescriptions } = {
+	boolean: assembleMemberDescriptions({
+		toString: returnsString,
+		valueOf: returnsBoolean
+	}),
+	number: assembleMemberDescriptions({
+		toExponential: returnsString,
+		toFixed: returnsString,
+		toLocaleString: returnsString,
+		toPrecision: returnsString,
+		toString: returnsString,
+		valueOf: returnsNumber
+	}),
+	string: assembleMemberDescriptions({
+		charAt: returnsString,
+		charCodeAt: returnsNumber,
+		codePointAt: returnsNumber,
+		concat: returnsString,
+		includes: returnsBoolean,
+		endsWith: returnsBoolean,
+		indexOf: returnsNumber,
+		lastIndexOf: returnsNumber,
+		localeCompare: returnsNumber,
+		match: returnsBoolean,
+		normalize: returnsString,
+		padEnd: returnsString,
+		padStart: returnsString,
+		repeat: returnsString,
+		search: returnsNumber,
+		slice: returnsString,
+		split: returnsArray,
+		startsWith: returnsBoolean,
+		substr: returnsString,
+		substring: returnsString,
+		toLocaleLowerCase: returnsString,
+		toLocaleUpperCase: returnsString,
+		toLowerCase: returnsString,
+		toString: returnsString,
+		toUpperCase: returnsString,
+		trim: returnsString,
```

Comment:
Maybe add `trimLeft` and `trimRight`.

Reply:
I wanted to but MDN was warning me about it:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimLeft

Reply:
Ah, ok sure, didn't see those.